src/Dapr.Actors/Client/ActorProxy.cs

```diff
@@ -75,7 +75,10 @@ public static TActorInterface Create<TActorInterface>(ActorId actorId, string ac
         /// <returns>Proxy to the actor object.</returns>
         public static object Create(ActorId actorId, Type actorInterfaceType, string actorType)
         {
-            _ = actorInterfaceType as IActor ?? throw new ArgumentException("The interface must implement IActor.", nameof(actorInterfaceType));
+            if(!typeof(IActor).IsAssignableFrom(actorInterfaceType))
+            {
+                throw new ArgumentException("The interface must implement IActor.", nameof(actorInterfaceType));
+            }
```

Comment:
Unit test? Obviously the success case here was untested

Reply:
added